reposerver/metrics/githandlers.go

```diff
@@ -19,7 +35,13 @@ func NewGitClientEventHandlers(metricsServer *MetricsServer) git.EventHandlers {
 		OnLsRemote: func(repo string) func() {
 			startTime := time.Now()
 			metricsServer.IncGitRequest(repo, GitRequestTypeLsRemote)
+			if lsRemoteParallelismLimitSemaphore != nil {
+				_ = lsRemoteParallelismLimitSemaphore.Acquire(context.Background(), 1)
```

Comment:
I think we are ignoring a possible error catch here. Should we be ignoring this?

Reply:
The `Acquire` method returns either `nil` or `ctx.Error()`. We are providing `context.Background()` which never get's canceled, so it is safe to ignore error here. Added comment that clarifies it.